pkg/sonar/componentService.go

```diff
@@ -16,17 +17,34 @@ const EndpointMeasuresComponent = "measures/component"
 type ComponentService struct {
 	Organization string
 	Project      string
+	Branch       string
 	apiClient    *Requester
 }
 
 type SonarCoverage struct {
-	Coverage       float32 `json:"coverage,omitempty"`
-	LineCoverage   float32 `json:"lineCoverage,omitempty"`
-	BranchCoverage float32 `json:"branchCoverage,omitempty"`
+	Coverage          float32 `json:"coverage"`
+	LineCoverage      float32 `json:"lineCoverage"`
+	LinesToCover      int     `json:"linesToCover"`
+	UncoveredLines    int     `json:"uncoveredLines"`
+	BranchCoverage    float32 `json:"branchCoverage"`
+	BranchesToCover   int     `json:"branchesToCover"`
+	UncoveredBranches int     `json:"uncoveredBranches"`
 }
 
-// GetCoverage ...
-func (service *ComponentService) Component(options *sonargo.MeasuresComponentOption) (*sonargo.MeasuresComponentObject, *http.Response, error) {
+type SonarLinesOfCode struct {
+	Total                int                         `json:"total"`
+	LanguageDistribution []SonarLanguageDistribution `json:"languageDistribution,omitempty"`
+}
+
+type SonarLanguageDistribution struct {
+	LanguageKey string `json:"languageKey,omitempty"` // Description:"key of the language as retrieved from sonarqube. All languages (key + name) are available as API https://<sonarqube-instance>/api/languages/list ",ExampleValue:"java,js,web,go"
+	LinesOfCode int    `json:"linesOfCode"`
+}
+
+func (service *ComponentService) Component(options *MeasuresComponentOption) (*sonargo.MeasuresComponentObject, *http.Response, error) {
+	if len(service.Branch) > 0 {
+		options.Branch = service.Branch
+	}
 	request, err := service.apiClient.create("GET", EndpointMeasuresComponent, options)
```

Comment:
The option `branch` is only available for SonarCloud and SonarQube Enterprise but not for SonarQube Community. Do you handle this somewhere?

Reply:
the option is only there in EE, but CE will still return the data from the default branch.
this is done in the same way for the issueService: https://github.com/SAP/jenkins-library/blob/master/pkg/sonar/issueService.go#L54

Reply:
Understood. The documentation on SonarQube Enterprise is just outdated for information about SonarQube Community